src/main/java/com/iexec/sms/tee/config/TeeWorkerInternalConfiguration.java

```diff
@@ -28,33 +26,38 @@
 import org.springframework.context.annotation.Configuration;
 import org.springframework.validation.annotation.Validated;
 
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
 @Slf4j
 @Configuration
 @Validated
 public class TeeWorkerInternalConfiguration {
-
     @Bean
     @ConditionalOnTeeFramework(frameworks = TeeFramework.GRAMINE)
-    GramineServicesProperties gramineServicesProperties(final TeeWorkerPipelineConfiguration pipelineConfig) {
-        final TeeWorkerPipelineConfiguration.Pipeline defaultPipeline = pipelineConfig.getPipelines().get(0);
-        final String frameworkVersion = defaultPipeline.version();
-        final TeeAppProperties preComputeProperties = defaultPipeline.preCompute().toTeeAppProperties();
-        final TeeAppProperties postComputeProperties = defaultPipeline.postCompute().toTeeAppProperties();
-
-        return new GramineServicesProperties(frameworkVersion, preComputeProperties, postComputeProperties);
+    public Map<String, TeeServicesProperties> gramineServicesPropertiesMap(
+            final TeeWorkerPipelineConfiguration pipelineConfig) {
+        return pipelineConfig.getPipelines().stream()
+                .map(pipeline -> pipeline.toTeeServicesProperties(null))
+                .collect(Collectors.toMap(
```

Comment:
Did I already mention I love immutability ?
```suggestion
                .collect(Collectors.toUnmodifiableMap(
```